gallery_dl/extractor/fantia.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://fantia.jp/"""
+
+from .common import Extractor, Message
+from .. import text
+
+
+class FantiaExtractor(Extractor):
+    """Base class for fantia extractors"""
+    category = "fantia"
+    root = "https://fantia.jp"
+    directory_fmt = ("{category}", "{fanclub_id}")
+    filename_fmt = "{post_id}_{file_id}.{extension}"
+    archive_fmt = "{post_id}_{file_id}"
+    _warning = True
+
+    def items(self):
+        yield Message.Version, 1
+
+        if self._warning:
+            if "_session_id" not in self.session.cookies:
+                self.log.warning("no '_session_id' cookie set")
+            FantiaExtractor._warning = False
+
+        for url, data in self.posts():
+            if data["content_filename"]:
+                fname = data["content_filename"]
+                data["filename"] = fname
+                if "." in fname:
+                    data["extension"] = fname[fname.rfind(".")+1:]
+                else:
+                    data["extension"] = text.ext_from_url(url)
```

Comment:
`filename` should not include the filename extension.
Why not `text.nameext_from_url(data["content_filename"] or url, data)`?